source/Information.h

```diff
@@ -49,6 +49,7 @@ class Information {
 
 	void SetCondition(const std::string &condition);
 	bool HasCondition(const std::string &condition) const;
+	void ClearConditions();
```

Comment:
I have the impression that this class was originally intended as an temporary "create it once, use it, then lose it" type of class.
Making part of the class re-usable seems quite a fundamental change and I'm a bit worried that this can give new problems in the future with outdated information showing up where we don't expect it.

Reply:
Maybe? That's why I added support for clearing the conditions, so the old code would work with minimal modifications. It is something that could cause issues with other open PRs that add interfaces, as they wouldn't be making this call. (That shouldn't be too hard to fix, and should be apparent from testing.)

Reply:
And other code will probably still re-create the Information part every frame, so that would likely not be an issue.

I was more worried about the possibility that some interfaces might have "outdated" information if this class gets re-used, but that might not be a big problem either, because also there if we encounter such issues, then we can just re-create the class every frame as is done now.